src/test/scala/com/microsoft/ml/spark/isolationforest/VerifyIsolationForest.scala

```diff
@@ -0,0 +1,137 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.ml.spark.isolationforest
+
+import java.io.File
+import java.nio.file.Files
+
+import com.microsoft.ml.spark.build.BuildInfo
+import com.microsoft.ml.spark.core.env.FileUtilities
+import com.microsoft.ml.spark.core.test.benchmarks.{Benchmarks, DatasetUtils}
+import org.apache.spark.ml.evaluation.BinaryClassificationEvaluator
+import org.apache.spark.ml.util.MLReadable
+import org.apache.spark.sql.{DataFrame, Dataset, Encoders, Row}
+import com.microsoft.ml.spark.core.test.fuzzing.{EstimatorFuzzing, TestObject}
+import org.apache.spark.TaskContext
+import org.apache.spark.ml.tuning.{CrossValidator, ParamGridBuilder}
+import org.apache.spark.sql.catalyst.encoders.RowEncoder
+import org.apache.spark.sql.functions._
+import org.apache.spark.ml.feature.VectorAssembler
+import org.apache.spark.ml.linalg.Vector
+import org.apache.spark.mllib.evaluation.BinaryClassificationMetrics
+import org.scalactic.Tolerance._
+
+case class LabeledDataPointVector(features: Vector, label: Double)
+case class MammographyRecord(feature0: Double, feature1: Double, feature2: Double, feature3: Double,
+                             feature4: Double, feature5: Double, label: Double)
+case class ScoringResult(features: Vector, label: Double, predictedLabel: Double, outlierScore: Double)
+case class ShuttleRecord(feature0: Double, feature1: Double, feature2: Double, feature3: Double,
+                         feature4: Double, feature5: Double, feature6: Double, feature7: Double,
+                         feature8: Double, label: Double)
+
+class VerifyIsolationForest extends Benchmarks with EstimatorFuzzing[IsolationForest] {
+  test ("Verify isolationForestMammographyDataTest") {
+    import session.implicits._
+
+    val data = loadMammographyData
+
+    // Train a new isolation forest model
+    val contamination = 0.02
+    val isolationForest = new IsolationForest()
+      .setNumEstimators(100)
+      .setBootstrap(false)
+      .setMaxSamples(256)
+      .setMaxFeatures(1.0)
+      .setFeaturesCol("features")
+      .setPredictionCol("predictedLabel")
+      .setScoreCol("outlierScore")
+      .setContamination(0.02)
+      .setContaminationError(contamination * 0.01)
+      .setRandomSeed(1)
+
+    // Score all training data instances using the new model
+    val isolationForestModel = isolationForest.fit(data)
+
+    // Calculate area under ROC curve and assert
+    val scores = isolationForestModel.transform(data).as[ScoringResult]
+    val metrics = new BinaryClassificationMetrics(scores.rdd.map(x => (x.outlierScore, x.label)))
```

Comment:
can we do this in the DF api?

Reply:
Can have a look. Copied from isolation forest repo

Reply:
BinaryClassificationMetrics wants an RDD as input... I don't think this looks nice: https://stackoverflow.com/questions/43941133/how-to-calculate-binary-classification-metrics-in-spark-mllib-with-dataframe-api

val scoreAndLabels = testResults.select("label", "probability")
    .rdd
    .map(row => 
            (row.getAs[Vector]("probability")(1), row.getAs[Double]("label"))
    )

Reply:
you can use com.microsoft.ml.spark.train.ComputeModelStatistics